example/ck_tile/01_fmha/fmha_fwd.cpp

```diff
@@ -507,90 +496,96 @@ bool run(const ck_tile::ArgParser& arg_parser)
+    }();
+    auto bias_host_view = (i_perm ? bias_host : bias_host.transpose(1, 2));
+    auto o_host_view    = (o_perm ? o_host : o_host.transpose(1, 2));
+
+    // verify result individually for each batch/group
     for(ck_tile::index_t wb = 0; wb < batch; ++wb)
     {
         const ck_tile::index_t real_seqlen_q = seqstart_q_host[wb + 1] - seqstart_q_host[wb];
         const ck_tile::index_t real_seqlen_k = seqstart_k_host[wb + 1] - seqstart_k_host[wb];
 
         // adjust matrix index according to the mode
-        const ck_tile::index_t b            = (mode == mode_enum::batch ? wb : 0);
-        const ck_tile::index_t query_offset = (mode == mode_enum::batch ? 0 : seqstart_q_host[wb]);
-        const ck_tile::index_t key_offset   = (mode == mode_enum::batch ? 0 : seqstart_k_host[wb]);
-
-        const auto v_host_ref_lengths =
-            std::array<ck_tile::index_t, 3>{nhead, hdim_v, real_seqlen_k};
-        const auto v_host_ref_strides =
-            is_v_rowmajor
-                ? std::array<ck_tile::index_t, 3>{hdim_v * real_seqlen_k, 1, hdim_v}
-                : std::array<ck_tile::index_t, 3>{hdim_v * real_seqlen_k, real_seqlen_k, 1};
-
-        ck_tile::HostTensor<QDataType> q_host_ref({nhead, real_seqlen_q, hdim_q});
-        ck_tile::HostTensor<KDataType> k_host_ref({nhead, real_seqlen_k, hdim_q});
-        ck_tile::HostTensor<VDataType> v_host_ref(v_host_ref_lengths, v_host_ref_strides);
-        ck_tile::HostTensor<ODataType> o_host_ref({nhead, real_seqlen_q, hdim_v});
+        const ck_tile::index_t b           = (mode == mode_enum::batch ? wb : 0);
+        const ck_tile::index_t query_start = (mode == mode_enum::batch ? 0 : seqstart_q_host[wb]);
+        const ck_tile::index_t query_end   = query_start + real_seqlen_q;
+        const ck_tile::index_t key_start   = (mode == mode_enum::batch ? 0 : seqstart_k_host[wb]);
+        const ck_tile::index_t key_end     = key_start + real_seqlen_k;
+        const ck_tile::index_t nr          = nhead / nhead_k;
+
+        // clang-format off
+        using Slice = ck_tile::HostTensorSlice;
+        auto q_host_view_slice = q_host_view
+                .index({Slice(0, b, b + 1), Slice(2, query_start, query_end)})
+                .squeeze(0);
+        auto k_host_view_slice = k_host_view
+                .index({Slice(0, b, b + 1), Slice(2, key_start, key_end)})
+                .squeeze(0)
+                .repeat({nr, 1, 1});
+        auto v_host_view_slice = v_host_view
+                .index({Slice(0, b, b + 1), Slice(3, key_start, key_end)})
+                .squeeze(0)
+                .repeat({nr, 1, 1});
+        auto o_host_view_slice = o_host_view
+                .index({Slice(0, b, b + 1), Slice(2, query_start, query_end)})
+                .squeeze(0);
+        // clang-format on
 
+        // create local tensors to speed-up computation
+        ck_tile::HostTensor<QDataType> q_host_ref(q_host_view_slice.get_lengths());
```

Comment:
Can we add clone() in HostTensorView?
ck_tile::HostTensor<QDataType> q_host_ref = q_host_view_slice.clone();

Reply:
`HostTensorView<> `don't know the existence of `HostTensor<>`. How can we create a subclass object from supoerclass?